### PR TITLE
Rename controller args and json keys

### PR DESCRIPTION
**Pull request by VicDeo**
Do some TODOs and fix one FIXME 

Basically renames 
`ldap_serverconfig_chooser` into `id` to match https://github.com/owncloud/user_ldap/pull/210
`ldap_clear_mapping` into `subject`

### CODE REVIEW

File: lib/Controller/MappingController.php [hunk 1]
```diff
@@ -62,12 +62,11 @@ public function __construct($appName,
...
 	 *
-	 * @param string $ldap_clear_mapping subject, 'group' or 'user'
-	 * @param string $ldap_serverconfig_chooser config id // FIXME remove in JS, is unneeded
+	 * @param string $subject, 'group' or 'user'
+	 *
 	 * @return DataResponse
 	 */
-	public function clear($ldap_clear_mapping, $ldap_serverconfig_chooser = null) {
-		$subject = $ldap_clear_mapping; // TODO if possible make JS send as 'subject' right away
+	public function clear($subject) {
```
Comment (jvillafanez) [inline lib/Controller/MappingController.php]: just to confirm the ldap_serverconfig_chooser / id isn't needed (?)

**Comment by jvillafanez** [comment]
Code-wise it looks fine, but I haven't tested it. Maybe @VicDeo can confirm it works without problems.

File: lib/Controller/MappingController.php (hunk 1 above)
Comment (VicDeo) [reply lib/Controller/MappingController.php]: no, it isn't. We truncate the whole user/group table https://github.com/owncloud/user_ldap/blob/master/lib/Mapping/AbstractMapping.php#L259-L264

**Comment by VicDeo** [comment]
@jvillafanez yes, I did a quick UI testing

**Pull request merged**